modules/expandable-panel/ExpandablePanel.tsx

```diff
@@ -0,0 +1,229 @@
+  }, [headerRef, windowSize]);
+
+  // calculate body height (once or every time window size changes)
+  useEffect(() => {
+    if (bodyHeight === 0 || bodyHeight === undefined) {
+      setBodyHeight(bodyRef.current?.getBoundingClientRect().height);
+    }
+  }, [bodyHeight, bodyRef, isExpanded, windowSize]);
+
+  const handleCollapse = useCallback(() => {
+    setCurrentBodyHeight(0);
+    setIsCollapsing(true);
+
+    // allow the animation to finish before removing the body from the DOM
+    setTimeout(() => {
+      setIsCollapsing(false);
+    }, SLIDE_ANIMATION_DURATION_MS);
+  }, []);
+
+  // set body height manually to be able to have the sliding animation
+  useEffect(() => {
+    if (isExpanded && bodyHeight && bodyHeight > 0 && currentBodyHeight === 0) {
+      setTimeout(() => {
+        setCurrentBodyHeight(bodyHeight);
+      }, 300);
+    }
+  }, [bodyHeight, currentBodyHeight, isExpanded]);
+
+  // state is being controlled by the parent component
+  useEffect(() => {
+    if (expanded !== undefined) {
+      if (!expanded) {
+        handleCollapse();
+      }
+
+      setIsExpanded(expanded);
+    }
+  }, [expanded, handleCollapse]);
+
+  const handleClick = () => {
+    if (disabled) {
+      return;
+    }
+
+    // state is being controlled internally
+    if (expanded === undefined) {
+      if (isExpanded) {
+        handleCollapse();
+      }
+
+      setIsExpanded(!isExpanded);
+    }
+
+    // eslint-disable-next-line mdx/no-unused-expressions
+    onClick?.();
+  };
+
+  const stateClasses = useMemo(() => {
+    const classes = [];
+
+    if (disabled) {
+      classes.push("disabled");
+    }
+
+    if (isExpanded) {
+      classes.push("expanded");
+    }
+
+    return classes;
+  }, [disabled, isExpanded]);
+
+  const renderHeader = useCallback(
+    () => (
+      <div ref={headerRef} className={cx(styles.header, ...stateClasses)}>
+        {typeof header === "string" ? (
+          <TextHighlight className={cx(styles.headerTitle)}>
+            {header}
+          </TextHighlight>
+        ) : (
+          header?.(isExpanded)
+        )}
+        <Icon
+          className={cx(styles.headerIcon, ...stateClasses)}
+          name="arrow"
+          size={16}
+        />
+      </div>
+    ),
+    [header, cx, styles, stateClasses, isExpanded]
+  );
+
+  const isBodyHeightReady = bodyHeight && bodyHeight > 0;
+  const canAnimate = isBodyHeightReady && (isExpanded || isCollapsing);
+  const finalBodyHeight = canAnimate ? currentBodyHeight : "100%";
+  const bodyWrapperStyles = {
+    height: finalBodyHeight,
+    maxHeight: finalBodyHeight === 0 && !isCollapsing ? 0 : undefined
+  };
+
+  return (
+    <div
+      className={cx(className, styles.panel, ...stateClasses)}
+      style={{ maxHeight: canAnimate ? "initial" : `${headerHeight}px` }}
+      onClick={handleClick}
+    >
+      {renderHeader()}
+      {(isExpanded || isCollapsing) && (
+        <div
+          ref={bodyWrapperRef}
```

Comment:
This ref isn't being used from what I can see right ? can we delete it? 

And is it possible to concat this div and the one below into one ? or this messes up with the animations ?

Reply:
Yes, you're right! I'll remove.

I purposefully added the wrapper div for the animation's sake, yes. In a nutshell, paddings were messing up the height calculation and animations, so this way I keep the body intact (height and padding and all), extract the correct height value for the animation from it, and animate the wrapper div